src/xtopcom/xdb/xdb_factory.h

```diff
@@ -19,10 +19,10 @@ class xdb_factory_t {
  public:
     xdb_factory_t() = delete;
     ~xdb_factory_t() = delete;
-    static std::shared_ptr<xdb_face_t> create(xdb_kind_t kind, const std::string& path);
-    static std::shared_ptr<xdb_face_t> instance(const std::string& path);
-    static std::shared_ptr<xdb_face_t> create_kvdb(const std::string& path) {
-        return create(xdb_kind_kvdb, path);
+    static std::shared_ptr<xdb_face_t> create(xdb_kind_t kind, const std::string& path, bool readOnly = false);
+    static std::shared_ptr<xdb_face_t> instance(const std::string& path, bool readOnly = false);
+    static std::shared_ptr<xdb_face_t> create_kvdb(const std::string& path, bool readOnly = false) {
+        return create(xdb_kind_kvdb, path, readOnly);
```

Comment:
![image](https://user-images.githubusercontent.com/2065762/123414605-b02f0d00-d5e6-11eb-9103-2611e3d632fb.png)
![image](https://user-images.githubusercontent.com/2065762/123414737-d8b70700-d5e6-11eb-8a73-1708cd0839a3.png)

the code itself proofs that only create_store_with_kvdb needs the default value.